pandas/_libs/tslibs/timedeltas.pyi

```diff
@@ -1,19 +1,65 @@
 from datetime import timedelta
 from typing import (
     ClassVar,
+    Literal,
     Type,
     TypeVar,
     overload,
 )
 
 import numpy as np
-import numpy.typing as npt
 
 from pandas._libs.tslibs import (
     NaTType,
     Tick,
 )
+from pandas._typing import npt
 
+# Copied from pandas/_libs/tslibs/timedeltas.pyx
+
+UnitChoices = Literal[
```

Comment:
Not sure whether this is possible/desired: Could de-duplicate this by declaring this Literal in _typing.py (not pyi!) and then import it in timedeltas.pyx and extract the possible values @jbrockmendel 
```py
>>> from typing import get_args, Literal
>>> get_args(Literal["a", "b", "c"])
('a', 'b', 'c')
```

Reply:
that sounds like a dependency-structure nightmare

Reply:
Another possibility is to just put this type in a new file in `pandas/typing/unitchoices.py`, and then `timedeltas.pyx` could import that instead, and `_typing.py` could import the new file as well.  That would remove the dependency problem.

Reply:
I'm not in a position to judge whether it is worth de-duplicating this long list.

If we have two copies, it would be good to have a comment with each copy to keep them in sync. Since `UnitChoices` is imported in a few files, might also be worth to move it to `_typing.py`.

Reply:
I looked more carefully at this.  In the `pyx` file, it is a `dict`.  We need the keys of that `dict`.  So we can't have the `pyx` code use the `Literal` from typing that comes from the keys in a `dict`.  I've put comments for cross-referencing in both files in  https://github.com/pandas-dev/pandas/pull/46343/commits/bf5c16ea28fcefebc49ba6d38240dfd400cb4d0f